overlord/configstate/config/transaction.go

```diff
@@ -287,7 +287,9 @@ func (t *Transaction) Commit() {
 	// Iterate through the write cache and save each item.
 	for instanceName, snapChanges := range t.changes {
 		config, ok := t.pristine[instanceName]
-		if !ok {
+		// due to LP #1917870 we might have a hook configure task in flight
+		// that tries to apply config over nil map, create it if nil.
+		if !ok || config == nil {
```

Comment:
is there a new test for this?

also if we need to do this I suppose we drop ok,  and just do config == nil